package/MDAnalysis/lib/mdamath.py

```diff
@@ -249,186 +250,6 @@ def box_volume(dimensions):
-
-    return True
-
-
-def make_whole(atomgroup, reference_atom=None):
-    """Move all atoms in a single molecule so that bonds don't split over images
-
-    Atoms are modified in place.
-
-    This function is most useful when atoms have been packed into the primary
-    unit cell, causing breaks mid molecule, with the molecule then appearing
-    on either side of the unit cell. This is problematic for operations
-    such as calculating the center of mass of the molecule. ::
-
-       +-----------+     +-----------+
-       |           |     |           |
-       | 6       3 |     |         3 | 6
-       | !       ! |     |         ! | !
-       |-5-8   1-2-| ->  |       1-2-|-5-8
-       | !       ! |     |         ! | !
-       | 7       4 |     |         4 | 7
-       |           |     |           |
-       +-----------+     +-----------+
-
-
-    Parameters
-    ----------
-    atomgroup : AtomGroup
-        The :class:`MDAnalysis.core.groups.AtomGroup` to work with.
-        The positions of this are modified in place.  All these atoms
-        must belong in the same molecule or fragment.
-    reference_atom : :class:`~MDAnalysis.core.groups.Atom`
-        The atom around which all other atoms will be moved.
-        Defaults to atom 0 in the atomgroup.
-
-    Raises
-    ------
-    NoDataError
-        There are no bonds present.
-        (See :func:`~MDAnalysis.topology.core.guess_bonds`)
-
-    ValueError
-        The algorithm fails to work.  This is usually
-        caused by the atomgroup not being a single fragment.
-        (ie the molecule can't be traversed by following bonds)
-
-    Note
-    ----
-    Only orthogonal boxes are currently supported.
-
-    Example
-    -------
-    Make fragments whole::
-
-        from MDAnalysis.util.mdamath import make_whole
-
-        # This algorithm requires bonds, these can be guessed!
-        u = mda.Universe(......, guess_bonds=True)
-
-        # MDAnalysis can split molecules into their fragments
-        # based on bonding information.
-        # Note that this function will only handle a single fragment
-        # at a time, necessitating a loop.
-        for frag in u.fragments:
-          make_whole(frag)
-
-    Alternatively, to keep a single atom in place as the anchor::
-
-        # This will mean that atomgroup[10] will NOT get moved,
-        # and all other atoms will move (if necessary).
-        make_whole(atomgroup, reference_atom=atomgroup[10])
-
-
-    .. versionadded:: 0.11.0
-    """
-    try:
-        b = atomgroup.bonds
-    except (AttributeError, NoDataError):
-        raise NoDataError("The atomgroup is required to have bonds")
-
-    if reference_atom is None:
-        ref = atomgroup[0]
-    else:
-        ref = reference_atom
-        # Sanity check
-        if not ref in atomgroup:
-            raise ValueError("Reference atom not in atomgroup")
-
-    # Check all of atomgroup is accessible from ref
-    if not _is_contiguous(atomgroup, ref):
```

Comment:
This test is not there in the new version. From what I understand, a non contiguous atom group will trigger the test at the very end of the new version, but the error message is not super clear about the reason, and it fails late instead of early.

Reply:
So when benchmarking this I found it takes a non negligible amount of time.  Because we also implicitly do this check when applying the algorithm, it made more sense to not check this before and just try.  We only apply the position change if the alg. worked, so it's defensive in that respect.  But yes this means that failing because of non-contiguous is slower, although succeeding is faster.

Reply:
OK. Good for me.